tools/run_tests/name_resolution/resolver_test_record_groups.yaml

```diff
@@ -0,0 +1,94 @@
+    - {TTL: '2100', data: 1.2.3.7, type: A}
+- expected_addrs: ['[2607:f8b0:400a:801::1001]:1234']
+  expected_config_index: null
+  record_to_resolve: srv-ipv6-single-target
+  records:
+    _grpclb._tcp.srv-ipv6-single-target:
+    - {TTL: '2100', data: 0 0 1234 ipv6-single-target, type: SRV}
+    ipv6-single-target:
+    - {TTL: '2100', data: '2607:f8b0:400a:801::1001', type: AAAA}
+- expected_addrs: ['[2607:f8b0:400a:801::1002]:1234', '[2607:f8b0:400a:801::1003]:1234',
+    '[2607:f8b0:400a:801::1004]:1234']
+  expected_config_index: null
+  record_to_resolve: srv-ipv6-multi-target
+  records:
+    _grpclb._tcp.srv-ipv6-multi-target:
+    - {TTL: '2100', data: 0 0 1234 ipv6-multi-target, type: SRV}
+    ipv6-multi-target:
+    - {TTL: '2100', data: '2607:f8b0:400a:801::1002', type: AAAA}
+    - {TTL: '2100', data: '2607:f8b0:400a:801::1003', type: AAAA}
+    - {TTL: '2100', data: '2607:f8b0:400a:801::1004', type: AAAA}
+- expected_addrs: ['1.2.3.4:1234']
+  expected_config_index: 0
+  record_to_resolve: srv-ipv4-simple-service-config
+  records:
+    _grpclb._tcp.srv-ipv4-simple-service-config:
+    - {TTL: '2100', data: 0 0 1234 ipv4-simple-service-config,
+      type: SRV}
+    ipv4-simple-service-config:
+    - {TTL: '2100', data: 1.2.3.4, type: A}
+    srv-ipv4-simple-service-config:
+    - {TTL: '2100', data: 'grpc_config=[{"serviceConfig":{"loadBalancingPolicy":"round_robin","methodConfig":[{"name":[{"method":"Foo","service":"SimpleService","waitForReady":true}]}]}}]',
+      type: TXT}
+- expected_addrs: ['1.2.3.4:443']
+  expected_config_index: 0
+  record_to_resolve: ipv4-no-srv-simple-service-config
+  records:
+    ipv4-no-srv-simple-service-config:
+    - {TTL: '2100', data: 1.2.3.4, type: A}
+    - {TTL: '2100', data: 'grpc_config=[{"serviceConfig":{"loadBalancingPolicy":"round_robin","methodConfig":[{"name":[{"method":"Foo","service":"NoSrvSimpleService","waitForReady":true}]}]}}]',
+      type: TXT}
+- expected_addrs: ['1.2.3.4:443']
+  expected_config_index: null
+  record_to_resolve: ipv4-no-config-for-cpp
+  records:
+    ipv4-no-config-for-cpp:
+    - {TTL: '2100', data: 1.2.3.4, type: A}
+    - {TTL: '2100', data: 'grpc_config=[{"clientLanguage":["python"],"serviceConfig":{"loadBalancingPolicy":"round_robin","methodConfig":[{"name":[{"method":"Foo","service":"PythonService","waitForReady":true}]}]}}]',
+      type: TXT}
+- expected_addrs: ['1.2.3.4:443']
+  expected_config_index: null
+  record_to_resolve: ipv4-cpp-config-has-zero-percentage
+  records:
+    ipv4-cpp-config-has-zero-percentage:
+    - {TTL: '2100', data: 1.2.3.4, type: A}
+    - {TTL: '2100', data: 'grpc_config=[{"percentage":0,"serviceConfig":{"loadBalancingPolicy":"round_robin","methodConfig":[{"name":[{"method":"Foo","service":"CppService","waitForReady":true}]}]}}]',
+      type: TXT}
+
+naming_end2end_tests_TODO:
+- "TODO: enable these large-txt-record tests once working. (these are txt-records are larger than 255 characters)"
```

Comment:
Yuchen, do you have time to look into this?  We don't need it to block this PR, but we will probably need to fix this before we can make c-ares the default DNS resolver.